### DISCUSSION TITLE
Running Arch Linux

### DISCUSSION THREAD

**Discussion by cemalgnlts** [copy/v86#851] [Q&A] [answered]
Hi,

I am trying to run the following example:
```js
var emulator = (window.emulator = new V86Starter({
			wasm_path: "/build/v86.wasm",
			memory_size: 512 * 1024 * 1024,
			vga_memory_size: 8 * 1024 * 1024,
			screen_container: document.getElementById("screen_container"),
			bios: {
				url: "/bios/seabios.bin",
			},
			vga_bios: {
				url: "/bios/vgabios.bin",
			},
			filesystem: {
				baseurl: "https://copy.sh/v86/images/arch/",
				basefs: "https://copy.sh/v86/images/fs.json",
			},
			autostart: true,
			bzimage_initrd_from_filesystem: true,
			cmdline: [
				"rw",
				"root=host9p rootfstype=9p rootflags=trans=virtio,cache=loose",
				"init=/usr/bin/init-openrc",
			].join(" "),
		}));
```

This is the output I got:
![Screenshot 2023-05-06 09 25 20](https://user-images.githubusercontent.com/45357531/236606206-d653867b-1e63-4f7e-969f-1e37ba939732.png)

**Comment by SuperMaxusa** [comment]
v86 can't get `fs.json` because the server is CORS enabled (https://github.com/copy/v86/discussions/822#discussioncomment-5600912) and it also throws a 404. You need to create 9pfs Arch Linux ([Archlinux guest setup](https://github.com/copy/v86/blob/master/docs/archlinux.md)) and host it on your local server.

https://github.com/copy/v86/blob/bbac74d589e2ea1de1ca2b434a68276101ebd722/docs/archlinux.md?plain=1#L393-L396

**Comment by cemalgnlts** [comment]
Could the problem be caused by something else. I can access linux.iso (https://copy.sh/v86/images/linux.iso).

Why `fs.json` cannot be accessed. 
Can I get the files I need in another way?

**Comment by cemalgnlts** [comment, marked as answer]
```
var emulator = (window.emulator = new V86Starter({
    screen_container: document.getElementById("screen_container"),
    wasm_path: "/build/v86.wasm",
    memory_size: 2000 * 1024 * 1024,
    autostart: true,
    initial_state: {
        url: "/arch_state.bin",
    },
}));
```

The problem was solved when I used the state I downloaded from the page by saying save state.
